src/componentWithName.js

```diff
@@ -16,7 +16,7 @@ function hasName(name, prop, propName, componentName) {
   }
 
   const { type } = prop;
-  if (getComponentName(type) !== name) {
+  if (!new RegExp(name).test(getComponentName(type))) {
```

Comment:
Okay I'll do that. Can you explain what might happen with an unsanitized string?